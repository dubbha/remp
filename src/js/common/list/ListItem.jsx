import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ListItem extends Component {
  static propTypes = {
    item: PropTypes.shape({
      poster: PropTypes.string.isRequired,
    }).isRequired,
    onSelectFilm: PropTypes.func.isRequired,
  };

  handleSelectFilm = () => {
    const { onSelectFilm, item } = this.props;

    onSelectFilm(item);
  }

  render() {
    const { item } = this.props;

    return (
      <div className="list__item">
        <div className="list__container">
          <a
            role="link"
            tabIndex={0}
            className="list__link"
            onClick={this.handleSelectFilm}
          >
            <img
              src={item.poster}
              alt={item.show_title}
              className="list__image"
            />
          </a>
          <div className="list__titleYearBlock">
            <a
              role="link"
              tabIndex={-1}
              className="list__link list__title"
              onClick={this.handleSelectFilm}
            >
              {item.show_title}
            </a>
            <span className="list__year">{item.release_year}</span>
          </div>
          <div className="list__category">{item.category}</div>
        </div>
      </div>
    );
  }
}
