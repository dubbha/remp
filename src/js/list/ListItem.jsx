import React from 'react';
import PropTypes from 'prop-types';

const ListItem = ({ item, onSelectFilm }) => (
  <div className="list__item">
    <div className="list__container">
      <img
        src={item.poster}
        className="list__image"
        onClick={() => onSelectFilm(item.show_id)} // eslint-disable-line react/jsx-no-bind
      />
      <div className="list__titleYearBlock">
        <span className="list__title">{item.show_title}</span>
        <span className="list__year">{item.release_year}</span>
      </div>
      <div className="list__category">{item.category}</div>
    </div>
  </div>
);

ListItem.propTypes = {
  item: PropTypes.shape({
    poster: PropTypes.string.isRequired,
  }).isRequired,
  onSelectFilm: PropTypes.func.isRequired,
};

export default ListItem;
