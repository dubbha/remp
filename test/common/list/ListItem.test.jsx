import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListItem from 'common/list/ListItem';

jest.mock('common/config/api.config', () => ({ imgUrl: 'https://IMAGE_URL/' }));

jest.mock('common/utils/parseYear', () => () => '2000');

jest.mock('common/utils/mapGenres', () => () => ['GENRE1', 'GENRE2', 'GENRE3']);

describe('ListItem', () => {
  let props;

  beforeEach(() => {
    props = {
      onSelectFilm: jest.fn(),
      item: {
        id: 12345678,
        title: 'TITLE',
        director: 'DIRECTOR',
        overview: 'OVERVIEW',
        poster_path: 'POSTER_PATH',
        release_date: '2000/12/31',
        runtime: 120,
        vote_average: 5.0,
        genre_ids: [1, 2, 3],
        cast: [
          { name: 'NAME1' },
          { name: 'NAME2' },
          { name: 'NAME3' },
        ],
      },
    };
  });

  it('should render successfully', () => {
    const wrapper = shallow(<ListItem {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handle film being selected on image click', () => {
    const wrapper = shallow(<ListItem {...props} />);

    wrapper.find('.list__link').at(0).simulate('click');

    expect(props.onSelectFilm).toBeCalled();
  });

  it('should handle film being selected on title click', () => {
    const wrapper = shallow(<ListItem {...props} />);

    wrapper.find('.list__link .list__title').simulate('click');

    expect(props.onSelectFilm).toBeCalled();
  });
});
