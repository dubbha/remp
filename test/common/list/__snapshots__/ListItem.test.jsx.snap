// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render successfully 1`] = `
<div
  className="list__item"
>
  <div
    className="list__container"
  >
    <a
      className="list__link"
      onClick={[Function]}
      role="link"
      tabIndex={0}
    >
      <img
        alt="TITLE"
        className="list__image"
        src="https://IMAGE_URL/POSTER_PATH"
      />
    </a>
    <div
      className="list__titleYearBlock"
    >
      <a
        className="list__link list__title"
        onClick={[Function]}
        role="link"
        tabIndex={-1}
      >
        TITLE
      </a>
      <span
        className="list__year"
      >
        2000
      </span>
    </div>
    <div
      className="list__category"
    >
      GENRE1
      GENRE2
      GENRE3
    </div>
  </div>
</div>
`;
